app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -48,35 +47,9 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
     // We crash if we init MediaSession at init time, hence lateinit.
     override lateinit var videoVoiceCommandMediaSession: VideoVoiceCommandMediaSession
 
-    private val sessionObserver by lazy {
-        object : SessionManager.Observer {
-            override fun onSessionSelected(session: Session) {
-                serviceLocator.screenController.showBrowserScreenForCurrentSession(supportFragmentManager, session)
-            }
-
-            override fun onSessionRemoved(session: Session) {
-                if (webRenderComponents.sessionManager.sessions.isEmpty()) {
-                    onNoActiveSession()
-                }
-            }
-
-            override fun onAllSessionsRemoved() {
-                onNoActiveSession()
-            }
-
-            private fun onNoActiveSession() {
-                // There's no active session. Start a new session with "homepage".
-                serviceLocator.screenController.showBrowserScreenForUrl(
-                    this@MainActivity,
-                    supportFragmentManager,
-                    URLs.APP_URL_HOME,
-                    Session.Source.NONE
-                )
-            }
-        }
-    }
-
     override fun onCreate(savedInstanceState: Bundle?) {
+        // We override onSaveInstanceState to not save state (for handling Clear Data), so startup flow
```

Comment:
We actually don't care if the state is restored or not, we just don't care about saving the state. I think the restoration of saved state still works fine, so that's not strictly necessary.